pkg/cli/server/root.go

```diff
@@ -736,12 +738,38 @@ func LoadConfiguration(config *config.Config, configPath string) error {
 	// we need another key delimiter.
 	viperInstance := viper.NewWithOptions(viper.KeyDelimiter("::"))
 
-	viperInstance.SetConfigFile(configPath)
+	ext := filepath.Ext(configPath)
+	ext = strings.ReplaceAll(ext, ".", "")
 
-	if err := viperInstance.ReadInConfig(); err != nil {
-		log.Error().Err(err).Msg("failed to read configuration")
+	/* if file extension is not supported, try everything
+	it's also possible that the filename is starting with a dot eg: ".config". */
+	if !common.Contains(viper.SupportedExts, ext) {
+		ext = ""
+	}
 
-		return err
+	switch ext {
+	case "":
+		log.Info().Msg("config file with no extension, trying all supported config types")
+
+		for _, configType := range viper.SupportedExts {
+			viperInstance.SetConfigType(configType)
+			viperInstance.SetConfigFile(configPath)
+
+			err := viperInstance.ReadInConfig()
+			if err == nil {
```

Comment:
Don't we need to track if:
- at least one extension there was no error, so the log at line 763 should not show at all
- all the extensions failed we should return early from this function
?